1.5.0 (#18)
btcpayserver/blog: blog/_posts/btcpay-server-1-5-0.md

---
title: "Introducing BTCPay Server 1.0.5 - Dashboard and payout processors"
date: 2022-04-28
author: pavlenex
category:
  - "releases"
tags:
  - "btcpay-server"
  - "release"
  - "payouts"
  - "dashboard"
  - "payout processors"
  - "update"
coverImage: "/images/BTCPay1-5-0-desk.png"
---

After a long wait, we're finally out of our coding cave, ready to introduce a new major BTCPay Server version - 1.5.0! 

> This release brings in a brand-new dashboard for easier analytics and better data visualization, payout processors and schedule for later for better payout management payments, batching your payments and refunds, several UX/UI improvements, greenfield API additions and a few non-critical bug fixes.

Reviewer: ```suggestion
This release brings in a brand-new dashboard for easier analytics and better data visualization, payout processors and schedule for later for better payout management payments, batching your payments and refunds, several UX/UI improvements, Greenfield API additions and a few non-critical bug fixes.
```

> This release brings in a brand-new dashboard for easier analytics and better data visualization, payout processors and schedule for later for better payout management payments, batching your payments and refunds, several UX/UI improvements, greenfield API additions and a few non-critical bug fixes.

Reviewer 2: ```suggestion
This release brings a brand-new dashboard for basic store analytics and better data visualization. But there is more to look out for in this release ; 
* Payout processor and 'Schedule for later' for better payout management.
* Batch payments and refunds. 
* Greenfield API additions
* Minor bug fixes 
* and several more UX/UI improvements! 
```

## New dashboard 📊

> We know that having a quick glance at your store's performance is essential. Also we understand that as a user, you would like to have an easy way to access and manage recent invoices and refunds. That's why we're introducing a **dashboard**!

Reviewer 2: ```suggestion
We know that having a quick glance at your store's performance is essential. We also heard you about easy access and management of recent invoices and refunds. This is where we introduce you to your new dashboard!
```

Author: Would commit, but this doesn't make sense

> This is where we introduce you to your new dashboard!

Probably meant `why`

Reviewer 2: ```suggestion
We know that having a quick glance at your store's performance is essential. We also heard you about easy access and management of recent invoices and refunds. With that in mind, we'd like to introduce you to your store's new **dashboard**!
``` 
No didn't mean Why, it makes sense to me as we already had the the Dashboard, but now we introduce them to a new layout (functionality) of that. But maybe it makes more sense this way :)

![](/images/1-5-0-dashboard-1.png)

![](/images/1-5-0-dashboard-2.png)

> Dashboard visually showcases critical store's data, providing you with a way to better understand your store's performance, so that you can make more informed decisions. Furthermore, a dashboard should help with easier management of recent invoices and pending payouts and refunds.

Reviewer 2: ```suggestion
Dashboard visually showcases critical store data, providing you with a way to better understand your store's performance to make more informed decisions. Furthermore, a dashboard should help manage recent invoices, pending payouts, and refunds more straightforwardly.
```

> Currently, the dashboard shows data for an on-chain payment method, pending payouts and refunds, recent invoices & transactions and the data for the crowdfunding app.

Reviewer 2: ```suggestion
Currently, the dashboard shows data for an on-chain payment method, pending payouts and refunds, recent invoices & transactions, and the data for the crowdfunding app.
```

> ![](/images/1-5-0-dashboard-3.png)

Reviewer 2: Can you made this lightmode Img ?  For consistency.

Author: The inconsistency has been added on purpose to showcase both modes.

> ![](/images/1-5-0-dashboard-4.png)

Reviewer 2: Can you made this lightmode Img ?  For consistency.

For more information on dashboard functionality, [check our docs](https://docs.btcpayserver.org/Dashboard/).

We're building the dashboard with flexibility in mind, and the goal is extend its functionality via the plugin system. 

> We'd love to hear what do you think about the dashboard and how it works for you. Let us know what kind of data would you help you better manage your store.

Reviewer 2: ```suggestion
We'd love to hear what you think about the dashboard and how it works for you. Let us know what kind of data would help you better manage your store
```

> ## Payout processor 🤖

Author: @Kukks need your input on technical accuracy and potential things that I might have missed.

> A payout processor is a new concept hooked into the [pull payment system], specifically the [payout part of it]. This new functionality allows you to automate the flow of the payouts.

Reviewer: ```suggestion
A payout processor is a new concept hooked into the [pull payment system](https://docs.btcpayserver.org/PullPayments/), specifically the [payout part of it](https://docs.btcpayserver.org/PullPayments/#approve-and-pay-a-payout). This new functionality allows you to automate the flow of the payouts.
```

> A payout processor is a new concept hooked into the [pull payment system](https://docs.btcpayserver.org/PullPayments/), specifically the [payout part of it](https://docs.btcpayserver.org/PullPayments/#approve-and-pay-a-payout). This new functionality allows you to automate the flow of the payouts.

Reviewer 3: ```suggestion
Payout processors are a new concept that hooks into the [pull payment system](https://docs.btcpayserver.org/PullPayments/), specifically the [payout part of it](https://docs.btcpayserver.org/PullPayments/#approve-and-pay-a-payout). This new functionality allows you to automate the flow of the payouts.
```

![](/images/1.5.0-payout-processor-1.png)

> Currently, we support payout processors for on-chain and off-chain payment methods. You can schedule payouts for on-chain, bolt11, lightning address, lnurl, etc.

Reviewer 3: ```suggestion
For this release, we've been two specific payout processors for on-chain and off-chain payment methods, meaning they support bitcoin addresses, BOLT11, lightning address, and LNURL.
```

Set up a time interval in your store settings upon which your payouts will be batched and broadcasted to the network. 

![](/images/1.5.0-payout-processor-2.png)

> Payout processors in a current shape and form allows you to save on transaction fees by batching the store's payments (wallet payments, refunds, payouts) and broadcasting them at a specific, pre-set time interval, for example, hourly, daily, weekly and more.

Reviewer 2: ```suggestion
Payout processor, in its current state, allows you to save on transaction fees by batching the store's payments (wallet payments, refunds, payouts) and broadcasting them at a specific, pre-set time interval, for example, hourly, daily, weekly and more.
```

> Payout processor, in its current state, allows you to save on transaction fees by batching the store's payments (wallet payments, refunds, payouts) and broadcasting them at a specific, pre-set time interval, for example, hourly, daily, weekly and more.

Reviewer 3: ```suggestion
Payout processors, in their current state, allow you to save on transaction fees by batching the store's payments (wallet payments, refunds, payouts) and broadcasting them at a specific, pre-set time interval, for example, hourly, daily, weekly, and more. Additionally, it reduces the management overhead and lets payouts to your customers and users be paid out automatically as soon as they provide their bitcoin payment request.
```

### Schedule for later 📆

> We understand that the payout processors may sound a bit abstract, and that's why we've added a concrete use-case for it through our brand-new **schedule for later** function in the wallet.

Reviewer 2: ```suggestion
We understand that 'payout processors' may sound a bit abstract. This is why we've built a concrete feature that comes with it, the brand-new schedule for later option in Wallet.
```

Reviewer 3: one sentence was better but like the quotation.

> We understand that the payout processors may sound a bit abstract, and that's why we've added a concrete use-case for it through our brand-new **schedule for later** function in the wallet.

Reviewer 3: ```suggestion
We understand that 'payout processors' may sound a bit abstract, and that's why we've added a concrete use-case for it through our brand-new **schedule for later** function in the wallet.
```

![](/images/1-5-0-schedule-transaction1.png)

This means that now, instead of signing and broadcasting a transaction, you have the option to schedule it for later on at a pre-set time interval.

> And since our refunding system goes through the wallet, the scheduling option now allows you to batch your customer's refunds and save on fees.

Reviewer 3: ```suggestion
And since our refunding system goes through the wallet, the scheduling option now allows you to automatically batch your customer's refunds and save on fees.
```

Reviewer 3: This didnt make sense since it was already possible

![](/images/1-5-0-schedule-transaction2.png)

> We plan to introduce a better permission system and hook it into the store so that in a particular store role, the store manager can create payouts, for example, refunds, but only the store owner can send them out.

Reviewer 3: ```suggestion
And since our refunding system goes through the pull payments system, if payout processors are configured on your store, they can now automatically be paid out.
```

Reviewer 3: This is already this way, I suggest we don't mention the store roles proposal just yet.

## Enable/disable users

> Server admins now have the options to disable or re-enable users, allowing them to better manage people who use their instance. 

Author: Merchants are just one type of users, so won't be modifying this. Also a typo server in BTCPay Server should be uppercase.

![](/images/1-5-0-enable-disable-users.png)

This functionality can be handy for third-party hosts or services building on top of BTCPay Server or offering BTCPay third-party hosting.

## NBXplorer updates 🔎

In case you've missed it, our dependency NBXPlorer has undergo a major performance refactoring. 

> Most of the refactoring was done to cather the way data is handled in the dashboard, but also to improve the performance for larger enterprise users.

Reviewer 2: ```suggestion
We did most of the refactoring in service of how data is handled in the dashboard and improved the performance for larger enterprise users.
```

You can read more about the NBXplorer updates [in this blog post](https://blog.btcpayserver.org/nbxplorer-postgres/).

## UX/UI improvements 🎨

As usually, with every release we're ironing out the user-interface, making incremental changes based on our understanding of how you use BTCPay Server. Your feedback is always critical factor when we're making such changes.

> - Better **lightning services access** - access lightning network services in a click of a button.

Reviewer: ```suggestion
- Better **Lightning services access** - access Lightning Network services with the click of a button.
```

Author: Is there really standardization for this? In Bitcoin Design we lowercase all to avoid confusion.

Reviewer: I mostly find it uppercased, as a specific term - e.g. the [Bitcoin wiki](https://en.bitcoin.it/wiki/Lightning_Network) and [Bitcoin Optech](https://bitcoinops.org/en/newsletters/2022/04/20/) refer to it as "Lightning Network".

> - **Store default currency for entities**, introduced in an earlier release, the default currency is now pre-set, but easily configurable when you're creating invoices, pull payments, pay button or payment requests.

Reviewer: ```suggestion
- **Store default currency for entities** - introduced in an earlier release, the default currency is now pre-set, but easily configurable when you're creating invoices, pull payments, pay button or payment requests.
```

- Add additional rate providers supported by CoinGecko.

## Greenfield API 👩‍💻

In this release we added a few new functionalities to our [Greenfield API](https://docs.btcpayserver.org/API/Greenfield/v1/).

* Add transaction info for on chain wallet 
* Add label filter for on chain wallet transaction

> * Add lightning payment info 

Reviewer: ```suggestion
* Add Lightning payment info 
```

> We're noticing an uptrend in people building on top of BTCPay Server. If you're building something with our Greenfield API, we're always looking to get more feedback on how to improve it.

Reviewer 2: ```suggestion
We're noticing an uptrend in people building on top of BTCPay Server. If you're building something with our Greenfield API, we're always looking to get more feedback on how to improve it and we love to see what you've made!
```

## Thank you 💚

As usual, if you have problems, feedback, feature requests regarding BTCPay Server, feel free to reach out on our [community chat](https://chat.btcpayserver.org/). We hope you enjoy what this update has to offer.

As always, thank you to our invaluable contributors, who directly participated in making 1.5.0 possible:

> [@dafunction](https://github.com/dafunction) [@dennisreimann](https://github.com/dennisreimann) [@dstrukt](https://github.com/dstrukt) [@Kukks](https://github.com/kukks/) [leesalminen](https://github.com/leesalminen/) [@NicolasDorier](https://github.com/nicolasdorier/)[orangesurf](https://github.com/(https://github.com/nicolasdorier/)/) [@ubolator](https://github.com/bolatovumar) [@woutersamaey](https://github.com/woutersamaey).

Reviewer: ```suggestion
[@dafunction](https://github.com/dafunction) [@dennisreimann](https://github.com/dennisreimann) [@dstrukt](https://github.com/dstrukt) [@Kukks](https://github.com/kukks/) [@leesalminen](https://github.com/leesalminen/) [@NicolasDorier](https://github.com/nicolasdorier/) [orangesurf](https://github.com/orangesurf) [@bolatovumar](https://github.com/bolatovumar) [@woutersamaey](https://github.com/woutersamaey).
```
